### PR TITLE
pytables-py: update to 3.5.1 and enable 3.7

### PR DESCRIPTION
**Pull request by kamischi**

### CODE REVIEW

**Comment by nieder** [comment]
tests fail immediately with:
````
Detected cores:      4
Default encoding:    utf-8
Default FS encoding: utf-8
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/sw/build.build/pytables-py37-3.5.1-1/PyTables-3.5.1/build/lib.macosx-10.13-x86_64-3.7/tables/tests/test_all.py", line 97, in test
    print_versions()
  File "/sw/build.build/pytables-py37-3.5.1-1/PyTables-3.5.1/build/lib.macosx-10.13-x86_64-3.7/tables/tests/common.py", line 147, in print_versions
    print('Default locale:      (%s, %s)' % locale.getdefaultlocale())
  File "/sw/lib/python3.7/locale.py", line 568, in getdefaultlocale
    return _parse_localename(localename)
  File "/sw/lib/python3.7/locale.py", line 495, in _parse_localename
    raise ValueError('unknown locale: %s' % localename)
ValueError: unknown locale: UTF-8
````
But adding `export LANG=en_US.UTF-8` in TestScript lets them run successfully:
....
Ran 6385 tests in 191.564s

OK (skipped=48)
TESTFAIL=$?
find /sw/build.build/pytables-py37-3.5.1-1/PyTables-3.5.1/build/lib.macosx-*-3.7 -name '*.py[oc]' -exec rm {} \;
exit $TESTFAIL
phase test: passed
.....

Line#30 has a typo: "Please install the %N-py27" <- unneeded '-py27'

pytables-py27 fails to build docs:
````
ls -d /sw/build.build/pytables-py27-3.5.1-1/PyTables-3.5.1/build/lib.macosx-*-2.7
Running Sphinx v1.8.4

Extension error:
Could not import extension IPython.sphinxext.ipython_console_highlighting (exception: No module named IPython.sphinxext.ipython_console_highlighting)
````
Should probably just turn off docs for now for all pyXX variants until sphinx gets its act together.

Most everything build/tests/installs fine on all python3X variants. It's still looking for -lblosc during configure. Is there a flag for forcing internal libblosc until? Not needed, but worth checking out.

Lastly, pyc cache files are still being left behind in `site-packages/tables`. See #324. I think PreRmScript needs to be fixed to clean those up.

**Comment by nieder** [comment]
One last thing I just found: needs a BuildDepends on setuptools-tng-py

**Comment by nieder** [comment]
Test failures when six-py is not installed (builds fine w/out it) but there are a lot of "import six" in the code, so maybe it's a Depends?

Also, I just committed libblosc from #220, so should add that to (Build)Depends to get consistent builds. That should be it.

**Comment by kamischi** [comment]
> Test failures when six-py is not installed (builds fine w/out it) but there are a lot of "import six" in the code, so maybe it's a Depends?
> 
> Also, I just committed libblosc from #220, so should add that to (Build)Depends to get consistent builds. That should be it.

Thanks for pointing out. I had overlooked the TestDepends on the already installed six-py.

**Comment by nieder** [comment]
It's a Depends for six-py in webob-py:
````
>>> import tables
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/sw/lib/python2.7/site-packages/tables/__init__.py", line 93, in <module>
    from .utilsextension import (
  File "tables/utilsextension.pyx", line 27, in init tables.utilsextension
  File "/sw/lib/python2.7/site-packages/tables/description.py", line 25, in <module>
    from . import atom
  File "/sw/lib/python2.7/site-packages/tables/atom.py", line 25, in <module>
    from .utils import SizeType
  File "/sw/lib/python2.7/site-packages/tables/utils.py", line 26, in <module>
    from .flavor import array_of_flavor
  File "/sw/lib/python2.7/site-packages/tables/flavor.py", line 54, in <module>
    from .exceptions import FlavorError, FlavorWarning
  File "/sw/lib/python2.7/site-packages/tables/exceptions.py", line 15, in <module>
    import six
ImportError: No module named six
````
pipdeptree also says that it needs mock-py, but I can import tables w/out mock-py installed:
````
* tables==3.5.1
 - mock [required: >=2.0, installed: ?]
````

**Comment by kamischi** [comment]
i do not really get, what needs to be done. Can you give some more details?

**Comment by nieder** [comment]
six-py needs to be in the Depends field, not TestDepends.
For mock-py, pipdeptree-py thinks pytables needs it, but pytables doesn't seem to actually need it. But upstream seems to think it is needed, so I would also add it to Depends just to be safe.

File: 10.9-libcxx/stable/main/finkinfo/sci/pytables-py.info [hunk 1]
```diff
@@ -21,13 +41,18 @@ CompileScript: <<
...
+  The PyTables HTML documentation currently cannot be built under Sphinx v1.8.4 due to
+  Sphinx errors.
+EOF
+#  export PYTHONPATH=$(ls -d %b/build/lib.macosx-*-%type_raw[python])
+#  make SPHINXBUILD=sphinx-build2.7 html
  else
   mkdir -p build/html
   cat <<EOF > build/html/README
   The PyTables HTML documentation currently cannot be built under Python3 due to
-  Sphinx errors. Please install the %N-py27 package for the offline docs.
```
Comment (dhomeier) [inline 10.9-libcxx/stable/main/finkinfo/sci/pytables-py.info]: The original motivation for this was that sphinx-py27 was working while the other variants were not.
With the docs build disabled entirely this does not make sense:
```
> cat /sw/share/doc/pytables-py36/html/README                                                                                          
  The PyTables HTML documentation currently cannot be built under Python3 due to
  Sphinx errors. Please install the pytables-py36 package for the offline docs.
```
so the last sentence should be removed or changed to `Please refer to https://www.pytables.org for the online docs.` In fact the entire test for `%type_pkg[python] -lt 30` can be removed as currently all variants are failing the same - the Sphinx `Extension error:
Could not import extension IPython.sphinxext.ipython_console_highlighting (exception: cannot import name 'path')` being due to a too old or missing ipython.
So in fact the whole doc build can be fixed by adding `ipython-py%type_pkg[python] (>= 5.7.0-1), sphinx-py%type_pkg[python] (>= 1.8.4-1)` to `BuildDepends`, but for that it seems I have to fix a number of conflicts in #164.

Comment (kamischi) [reply 10.9-libcxx/stable/main/finkinfo/sci/pytables-py.info]: I was simply too hesitating to remove stuff from a package description of someone else. As you think, this is fine, i did so in commit c579b0d.

**Comment by nieder** [comment]
Looks fine. Will wait for @dhomeier to weigh in since it's his.

**Comment by dhomeier** [comment]
Keeping the doc build instructions as a reminder for the time when we will have ipython 5 in principle is not a bad idea, but as it's stored in the git history, having a clean .info now probably is better.

**Pull request merged**